### DISCUSSION TITLE
being able to retain highlight from a Word document?

### DISCUSSION THREAD

**Discussion by lstockton** [ueberdosis/tiptap#3131] [General Questions]
I'm new to Tiptap and so far I like it.  I've been able to get quite a bit of functionality out of it in the first day of use.  One thing that I was hoping for, that it doesn't do for me (yet) is that if a user copies from a Word document and pastes into the Taptap editor, it sure would be nice to be able to retain any highlighting in the text.
So far, I can author highlighting from within Tiptap itself, but not from pasting from Word.

**Comment by Priestch** [comment]
@lstockton tiptap use prosemirror mark to implement highlight. You can try to use [transformPastedHTML](https://prosemirror.net/docs/ref/#view.EditorProps.transformPastedHTML) to transform the content copied from a Word document to conform to what is expected by tiptap.

**Reply by lstockton** [reply]
I'm very new to Tiptap.  and I don't know much about ProseMirror, but I want to do this.  I appreciate that they've provided a way.  I just don't know how to do it.
To implement this in Tiptap, what do I have to do to get to transformPastedHTML?  I can't find a single example of how to set this up.

> const Tiptap = ({ setUpdateDescription }) => {
  const editor = useEditor({
    extensions: [StarterKit, Underline, Highlight.configure({ multicolor: false }),
      TextStyle, Color, FontFamily.configure({ types: ['textStyle'], })],
    content: "<p></p>",
    onUpdate: ({ editor }) => {
      const html = editor.getHTML();
      setUpdateDescription(html);
    },
  });

> 

so, is there a thing I put in after that last comma, to specify that use of transformPastedHTML?

**Reply by gethari** [reply]
@lstockton no it doesn't work like that . Try to look into other editor repositories by looking using the tags in GitHub. You might end up finding a gold mine

**Reply by lstockton** [reply]
I have searched and searched, and I can't find anything that shows me how to tie this into Tiptap.

**Reply by Priestch** [reply]
@lstockton Check [Prosemirror Plugins](https://tiptap.dev/guide/custom-extensions#prosemirror-plugins)

**Reply by lstockton** [reply]
I struggled but got through it.  thanks
